### ISSUE TITLE
Line/fill oddities with large polygon

### ISSUE THREAD

**Issue by pnorman** [mapnik/mapnik#3589]
A large node move in OSM revealed some kind of bug in Mapnik, so I'm opening this issue for tracking as we figure out more and reduce it to a testcase.

![image](https://cloud.githubusercontent.com/assets/1190866/21668517/6160af08-d2b7-11e6-91d3-15eb41dafc45.png)

The way in question was

![image](https://cloud.githubusercontent.com/assets/1190866/21668549/c41b5634-d2b7-11e6-84d1-1ac5f1de05a1.png)

With an overall shape of 
![image](https://cloud.githubusercontent.com/assets/1190866/21668556/d33cf3d4-d2b7-11e6-93ef-7f836f064a00.png)

I am confident that there is a bug in Mapnik, not osm2pgsql, because the outer line colour appeared over a huge area. An osm2pgsql bug would have resulted in a large object drawn normally, not the wrong colour filling the object.

On the NW corner of the bounding box, there was

![image](https://cloud.githubusercontent.com/assets/1190866/21668593/2aa8128e-d2b8-11e6-845b-afdac1768f98.png)
 The west edge happened to lie on a tile edge, but the north didn't. The colour here is the line colour, not the fill colour, which is weird.

cc @nicolas17

**Comment by flippmoke** [comment]
Is this from a vector tile? Perhaps the vector tile has some strange coordinate within it?

**Comment by pnorman** [comment]
> Is this from a vector tile?

Nope, standard Mapnik style. The polygon bbox intersects with the rendered area in examples like the NW corner so it gets returned from the datasource.

**Comment by artemp** [comment]
@pnorman - this looks like known issue with numbers getting far too large when converted into (meta)tile coordinate system. One way to handle this is to enable clipping but for buildings it doesn't really makes sense. Can you post offending polygon in GeoJSON or WKT here so I can investigate further, pls ?
